controllers/useraccount/useraccount_controller.go

```diff
@@ -506,66 +506,56 @@ func (r *Reconciler) deleteIdentityAndUser(logger logr.Logger, config membercfg.
 	return false, nil
 }
 
-// deleteUser deletes the user resource.
-// Returns `true` if the user was deleted, `false` otherwise, with the underlying error
-// if the user existed and something wrong happened. If the user did not exist,
+// deleteUser deletes the user resources associated with the specified UserAccount.
+// Returns `true` if the users were deleted, `false` otherwise, with the underlying error
+// if the user existed and something wrong happened. If the users don't exist,
 // this func returns `false, nil`
 func (r *Reconciler) deleteUser(logger logr.Logger, userAcc *toolchainv1alpha1.UserAccount) (bool, error) {
-	// Get the User associated with the UserAccount
-	user := &userv1.User{}
-	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name}, user)
+	userList := &userv1.UserList{}
+	err := r.Client.List(context.TODO(), userList, listByOwnerLabel(userAcc.Name))
 	if err != nil {
-		if !errors.IsNotFound(err) {
-			return false, err
-		}
-		return false, nil
+		return false, err
 	}
-	if _, exist := user.Labels[toolchainv1alpha1.OwnerLabelKey]; !exist {
-		logger.Info("the User resource wasn't created by Toolchain Member operator - skipping", "user", userAcc.Name)
+
+	if len(userList.Items) == 0 {
 		return false, nil
 	}
 
-	logger.Info("deleting the User resource")
-	// Delete User associated with UserAccount
-	if err := r.Client.Delete(context.TODO(), user); err != nil {
-		if !errors.IsNotFound(err) {
+	logger.Info("deleting the User resources")
+	for _, user := range userList.Items {
+		// Delete User associated with UserAccount
+		if err := r.Client.Delete(context.TODO(), &user); err != nil {
```

Comment:
This controller watches users, so shouldn't we delete one user form the list and then return? The deletion of the user will trigger another reconcile.
We should not delete multiple Users (or Identities) in the same reconcile loop.